### PR TITLE
nitc: check pkg-config packages availability later

### PR DESCRIPTION
**Pull request by xymus**
The `pkgconfig` annotation tells the compiler to use the `pkg-config` command in order to get the C compiler and linker options required to build the user C code. The availability of each package was tested twice, once at reading the annotation and once in the generated Makefile. This PR moves the first check later, with the generation of the Makefile. There is still two checks, one by the Nit compiler with a prettier output, and one by the Makefile in case it is distributed with the generated C source files.

This fixes an issue we've had when compiling for Android and iOS, where the result of the annotation was not used but still blocked the compilation if the host did not have the package. And leaving the check to the Makefile give a chance to the programmer to tweak it before the C compilation when cross-compiling.

The error message thrown by the Makefile is less user-friendly as we lose the reference to the Nit source file and line number. This should not be much on an issue since it is rarely a programming error, more of a system configuration error, but it could be improved upon if it becomes an issue.

### CODE REVIEW

**Comment by xymus** [comment]
This causes a new dirty workspace error on the tests server. It will be fixed by #2611.

File: src/compiler/abstract_compiler.nit [hunk 1]
```diff
@@ -491,6 +516,7 @@ endif
 		end
 
 		# Compile each required extern body into a specific .o
+		var java_files = new Array[ExternFile]
```
Comment (privat) [inline src/compiler/abstract_compiler.nit]: th move of java_files seems unrelated with the commit

File: src/ffi/pkgconfig.nit [hunk 2]
```diff
@@ -67,27 +67,7 @@ private class PkgconfigPhase
 			end
 		end
 
-		# check availability of pkg-config
```
Comment (privat) [inline src/ffi/pkgconfig.nit]: To keep concern together, instead of cutnpaste, why not keeping the code in `pkgconfig.nit` in a specific function; then calling this new function from `abstract_compiler.nit`?

Comment (privat) [reply src/ffi/pkgconfig.nit]: this specific function may also be called at different time: earlier on linux, latter when crosscomiling...

File: src/compiler/abstract_compiler.nit [hunk 3]
```diff
@@ -188,6 +188,8 @@ class MakefileToolchain
 		var time1 = get_time
 		self.toolcontext.info("*** END WRITING C: {time1-time0} ***", 2)
 
+		if not toolcontext.check_errors then return
```
Comment (privat) [inline src/compiler/abstract_compiler.nit]: why do we need that here by the way?

Comment (xymus) [reply src/compiler/abstract_compiler.nit]: To stop if errors on missing pkg-config packages were raised by `write_makefile`, before the C compilation.

Comment (xymus) [reply src/compiler/abstract_compiler.nit]: `write_makefile` may not be the best place for the pkg-config check. In fact, the whole check logic could be moved somewhere else, but it needs to know the mainmodule and the target platform, and having the check after the generation of the C code is useful but not vital.

File: src/compiler/abstract_compiler.nit (hunk 1 above)
Comment (xymus) [reply src/compiler/abstract_compiler.nit]: Maybe, but it was obviously at the wrong place when I added the documentation where it was previously.

File: src/ffi/pkgconfig.nit (hunk 2 above)
Comment (xymus) [reply src/ffi/pkgconfig.nit]: The real difference between the two checks in Nit code (the one in nitc and the one in niti) is the way errors are raised. Can we replace the use of `NaiveInterpreter::fatal` by `ToolContext::error`? Or insert a `check_error` afterwards?

Comment (privat) [reply src/ffi/pkgconfig.nit]: I agree. factorizing the check with the one in niti is the way to go

**Comment by xymus** [comment]
I fixed the duplicated pkg-config checks as requested. It makes `abstract_compiler` depend directly on the `pkgconfig` module. I could move up the method declaration to remove this dependancy and keep nith lighter, but this would add complexity.
